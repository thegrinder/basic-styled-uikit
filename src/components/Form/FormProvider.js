import { createProvider } from '../../helpers';

const FormProvider = createProvider('form', {});

export default FormProvider;
