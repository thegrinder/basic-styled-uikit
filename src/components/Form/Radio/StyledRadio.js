import styled from 'styled-components';
import { bool } from 'prop-types';
import { miscInputStyles } from '../commonFormStyles';
import { rem } from '../../../theme/typography';

const propTypes = {
  invalid: bool.isRequired,
};

const StyledRadio = styled.input.attrs({
  type: 'radio',
})`
  ${miscInputStyles}
  border-radius: 50%;
  position: relative;
  &:before {
    content: '';
    position: absolute;
    top: 0;
    bottom: 0;
    left: 0;
    right: 0;
    margin: auto;
    width: ${rem(6)};
    height: ${rem(6)};
    border-radius: 50%;
  }
  &:checked {
    &:before {
      background-color: #fff;
    }
  }
`;

StyledRadio.propTypes = propTypes;

export default StyledRadio;
