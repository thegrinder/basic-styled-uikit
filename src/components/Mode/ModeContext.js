import { createContext } from 'react';

const ModeContext = createContext(undefined);

export default ModeContext;
