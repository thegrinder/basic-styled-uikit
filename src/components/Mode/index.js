export { default as ModeProvider } from './ModeProvider';
export { default as useMode } from './useMode';
