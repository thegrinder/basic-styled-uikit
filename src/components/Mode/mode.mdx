---
name: Mode
menu: Components
route: /components/mode
---

## **ModeProvider**



---



## **useMode**
