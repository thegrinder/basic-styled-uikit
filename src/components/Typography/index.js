export { default as TypographyProvider } from './TypographyProvider';
export { default as Text } from './Text/Text';
export * from './typographySelectors';
