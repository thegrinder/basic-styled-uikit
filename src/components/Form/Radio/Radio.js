import React from 'react';
import { bool } from 'prop-types';
import StyledRadio from './StyledRadio';

const propTypes = {
  /** invalid flag */
  invalid: bool,
};

const Radio = ({ invalid = false, ...rest }) => (
  <StyledRadio invalid={invalid} {...rest} />
);

Radio.propTypes = propTypes;

export default Radio;
