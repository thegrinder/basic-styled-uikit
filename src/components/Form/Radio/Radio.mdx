---
name: Radio
menu: Components
---

import { Playground, Props } from 'docz';
import Radio from './Radio';

# Radio

<Props of={Radio} />

### Radio

<Playground>
  <Radio />
</Playground>

### Invalid Radio

<Playground>
  <Radio invalid={true} />
</Playground>

### Disabled Radio

<Playground>
  <Radio disabled={true} />
  <Radio checked={true} disabled={true} />
</Playground>
