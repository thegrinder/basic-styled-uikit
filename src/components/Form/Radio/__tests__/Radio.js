import React from 'react';
import { render } from '@testing-library/react';
import { ModeProvider } from '../../../Mode';
import { TypographyProvider } from '../../../Typography';
import FormProvider from '../../FormProvider';
import Radio from '../Radio';

const testId = 'radio';

const renderComponent = (props = {}) =>
  render(
    <ModeProvider>
      <TypographyProvider>
        <FormProvider>
          <Radio data-testid={testId} {...props} />
        </FormProvider>
      </TypographyProvider>
    </ModeProvider>
  );

describe('<Radio />', () => {
  it('should render correctly with default props', () => {
    const { queryByTestId } = renderComponent();
    expect(queryByTestId(testId)).toBeTruthy();
  });

  it('should render correctly with custom props', () => {
    const { queryByTestId } = renderComponent({
      invalid: true,
    });
    expect(queryByTestId(testId)).toBeTruthy();
  });

  it('should render <input> tag with type radio', () => {
    const { queryByTestId } = renderComponent();
    expect(queryByTestId(testId).tagName).toEqual('INPUT');
    expect(queryByTestId(testId).type).toEqual('radio');
  });
});
