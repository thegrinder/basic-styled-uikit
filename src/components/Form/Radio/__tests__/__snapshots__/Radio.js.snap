// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Radio /> should render correctly with custom props 1`] = `
<input
  class="StyledRadio-sc-1q7ubkv-0 kGcskO"
  type="radio"
/>
`;

exports[`<Radio /> should render correctly with default props 1`] = `
<input
  class="StyledRadio-sc-1q7ubkv-0 liLDQG"
  type="radio"
/>
`;
